### PR TITLE
logging: hide entries when debuglog is set to false and add missing worker names (PROJQUAY-6562)

### PR DESCRIPTION
**Pull request by sayalibhavsar**
1. Stop log entries from the util.migrate.allocator.py module from being displayed in Quay logs when the debug log is false.
2. Only show them when the debug log is set to true with the name of the worker printing them

### CODE REVIEW

**Comment by sayalibhavsar** [comment]
Hello @dongboyan77 tested it in dev and it looks fine. Can you take a look?
cc. @Marcusk19 @HammerMeetNail 

1. (myenv) sbhavsar@sbhavsar-systemversion:~/myquay$ cat docker-compose.yaml | grep DEBUGLOG
DEBUGLOG: "true"

(myenv) sbhavsar@sbhavsar-systemversion:~/myquay$ cat quay-debug.op | grep util.migrate.allocator
securityworker stdout | 2024-08-23 11:38:05,784 [314] [DEBUG] [util.migrate.allocator] Total range: 1-2
securityworker stdout | 2024-08-23 11:38:05,785 [314] [DEBUG] [util.migrate.allocator] Selected random hole 0 with 1 total holes
securityworker stdout | 2024-08-23 11:38:05,785 [314] [DEBUG] [util.migrate.allocator] Selecting from hole range: 1-2
securityworker stdout | 2024-08-23 11:38:05,785 [314] [DEBUG] [util.migrate.allocator] Rand max bound: 1
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] No candidates, marking entire block completed 1-2 by worker 
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] Marking the range completed: 1-2
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] Discarding block and setting new max to: 1
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] Discarding block and setting new min to: 2
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] Total blocks: 0
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] Total range: 2-1
securityworker stdout | 2024-08-23 11:38:05,788 [314] [DEBUG] [util.migrate.allocator] No more work by worker 


2. (myenv) sbhavsar@sbhavsar-systemversion:~/myquay$ cat docker-compose.yaml | grep DEBUGLOG
DEBUGLOG: "false"

(myenv) sbhavsar@sbhavsar-systemversion:/myquay$ cat quay-info.op | grep util.migrate.allocator
(myenv) sbhavsar@sbhavsar-systemversion:/myquay$

File: util/migrate/allocator.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 import random
 from threading import Event
 
 from bintrees import RBTree
 
 logger = logging.getLogger(__name__)
-logger.setLevel(logging.DEBUG)  # Set the default log level to DEBUG
+
+# Read the DEBUGLOG environment variable
+debug_log = os.getenv("DEBUGLOG", "false").lower() == "true"
```
Comment (michaelalang) [inline util/migrate/allocator.py:11]: wouldn't a boolean check be more flexible to ensure it doesn't matter how one writes `enable` ?

```
from pydantic import TypeAdatper
debug_log = TypeAdapter(bool).validate_python(
                          os.environ.get("DEBUGLOG", "False"))
```

that would give us 
```
>>> for val in (True, "True", 1, "yes", "Yes", False, "False", 0, "no", "NO"):
...        print(f"value {val} evaluated to {TypeAdapter(bool).validate_python(val)}")
... 
value True evaluated to True
value True evaluated to True
value 1 evaluated to True
value yes evaluated to True
value Yes evaluated to True
value False evaluated to False
value False evaluated to False
value 0 evaluated to False
value no evaluated to False
value NO evaluated to False
```

**Review by bcaton85 — APPROVED** [review]
LGTM but we should let QE know about the upgrade to the `PyMySQL` package. This could cause issues

**Comment by ibazulic** [comment]
I think this breaks logging a bit during startup:

~~~
Running init script '/quay-registry/conf/init/client_certs.sh'
Running init script '/quay-registry/conf/init/copy_config_files.sh'
Running init script '/quay-registry/conf/init/d_validate_config_bundle.sh'
Validating Configuration
time="2024-08-30T20:09:35Z" level=debug msg="Validating AccessSettings"
time="2024-08-30T20:09:35Z" level=debug msg="Validating ActionLogArchiving"
time="2024-08-30T20:09:35Z" level=debug msg="Validating AppTokenAuthentication"
time="2024-08-30T20:09:35Z" level=debug msg="Validating AutoPrune"
...
DEBUG:app:Loading default config.
DEBUG:util.config.provider.basefileprovider:Applying config file: /quay-registry/conf/stack/config.yaml
DEBUG:app:Loaded config
INFO:util.ipresolver:Loading AWS IP ranges from disk
DEBUG:util.ipresolver:Building AWS IP ranges
DEBUG:util.ipresolver:Finished building AWS IP ranges
DEBUG:botocore.hooks:Changing event name from creating-client-class.iot-data to creating-client-class.iot-data-plane
DEBUG:botocore.hooks:Changing event name from before-call.apigateway to before-call.api-gateway
DEBUG:botocore.hooks:Changing event name from request-created.machinelearning.Predict to request-created.machine-learning.Predict
DEBUG:botocore.hooks:Changing event name from before-parameter-build.autoscaling.CreateLaunchConfiguration to before-parameter-build.auto-scaling.CreateLaunchConfiguration
DEBUG:botocore.hooks:Changing event name from before-parameter-build.route53 to before-parameter-build.route-53
...
DEBUG:peewee.pool:Created new connection 123867728399872.
DEBUG:peewee:('SELECT "t1"."id", "t1"."name" FROM "imagestoragelocation" AS "t1" WHERE ("t1"."name" IN (%s))', ['minio-1'])
DEBUG:peewee:('DELETE FROM "servicekey" WHERE ((("servicekey"."service" = %s) AND ("servicekey"."expiration_date" <= %s)) OR ((("servicekey"."service" = %s) AND ("servicekey"."approval_id" IS %s)) AND ("servicekey"."created_date" <= %s)))', ['quay', datetime.datetime(2024, 8, 23, 20, 9, 41, 359689), 'quay', None, datetime.datetime(2024, 8, 29, 20, 9, 41, 359705)])
DEBUG:peewee:('INSERT INTO "servicekey" ("name", "kid", "service", "jwk", "metadata", "created_date", "expiration_date", "rotation_duration") VALUES (%s, %s, %s, %s, %s, %s, %s, %s) RETURNING "servicekey"."id"', ['https://quay.skynet:443', 'uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU', 'quay', '{"n": "3NeZu9...", "e": "AQAB", "kty": "RSA", "kid": "uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU"}', '{"created_by": "CLI tool"}', datetime.datetime(2024, 8, 30, 20, 9, 41, 360790), datetime.datetime(2024, 8, 30, 22, 9, 41, 297722), None])
DEBUG:peewee:('SELECT "t1"."id", "t1"."uuid", "t1"."username", "t1"."password_hash", "t1"."email", "t1"."verified", "t1"."stripe_id", "t1"."organization", "t1"."robot", "t1"."invoice_email", "t1"."invalid_login_attempts", "t1"."last_invalid_login", "t1"."removed_tag_expiration_s", "t1"."enabled", "t1"."invoice_email_address", "t1"."given_name", "t1"."family_name", "t1"."company", "t1"."location", "t1"."maximum_queued_builds_count", "t1"."creation_date", "t1"."last_accessed" FROM "user" AS "t1" WHERE ("t1"."username" IN (%s, %s))', ['admin', 'ibazulic'])
DEBUG:peewee:('SELECT "t1"."id", "t1"."name" FROM "notificationkind" AS "t1" WHERE ("t1"."name" = %s) LIMIT %s OFFSET %s', ['service_key_submitted', 1, 0])
DEBUG:peewee:('INSERT INTO "notification" ("uuid", "kind_id", "target_id", "metadata_json", "created", "dismissed", "lookup_path") VALUES (%s, %s, %s, %s, %s, %s, %s) RETURNING "notification"."id"', ['944e4b98-1172-4ab9-b115-86b8a0a2706e', 11, 1, '{"name": "https://quay.skynet:443", "kid": "uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU", "service": "quay", "jwk": {"n": "3NeZu9EN..."}, "metadata": {"created_by": "CLI tool"}, "created_date": 1725048581, "expiration_date": 1725055781}', datetime.datetime(2024, 8, 30, 20, 9, 41, 371392), False, '/service_key_approval/uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU/1'])
DEBUG:peewee:('SELECT "t1"."id", "t1"."name", "t1"."kid", "t1"."service", "t1"."jwk", "t1"."metadata", "t1"."created_date", "t1"."expiration_date", "t1"."rotation_duration", "t1"."approval_id" FROM "servicekey" AS "t1" WHERE ("t1"."kid" = %s) LIMIT %s OFFSET %s FOR UPDATE', ['uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU', 1, 0])
DEBUG:peewee:('INSERT INTO "servicekeyapproval" ("approver_id", "approval_type", "approved_date", "notes") VALUES (%s, %s, %s, %s) RETURNING "servicekeyapproval"."id"', [None, 'ServiceKeyApprovalType.AUTOMATIC', datetime.datetime(2024, 8, 30, 20, 9, 41, 375454), ''])
DEBUG:peewee:('UPDATE "servicekey" SET "name" = %s, "kid" = %s, "service" = %s, "jwk" = %s, "metadata" = %s, "created_date" = %s, "expiration_date" = %s, "rotation_duration" = %s, "approval_id" = %s WHERE ("servicekey"."id" = %s)', ['https://quay.skynet:443', 'uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU', 'quay', '{"n": "3NeZ..."}', '{"created_by": "CLI tool"}', datetime.datetime(2024, 8, 30, 20, 9, 41, 360790), datetime.datetime(2024, 8, 30, 22, 9, 41, 297722), None, 8, 8])
DEBUG:peewee:('DELETE FROM "notification" WHERE ("notification"."lookup_path" ILIKE %s)', ['/service_key_approval/uRzu6i1fDXD8XEHofVBAtl2RbmB5Rq7huIb1og-mYeU%'])
...
~~~

and then logs continue normally. Do you think we can fix this somehow? Some better formatting perhaps?

**Pull request merged**